nixos/tests/swap-file-btrfs.nix

```diff
@@ -32,6 +32,8 @@ import ./make-test-python.nix ({ lib, ... }:
 
   testScript = ''
     machine.wait_for_unit('var-swapfile.swap')
+    # Ensure the swap file creation script ran to completion without failing when creating the swap file
+    machine.fail("systemctl is-failed --quiet mkswap-var-swapfile.service")
```

Comment:
Is there a simpler way to test the unit did not exit with a failure than this awkward double negative logic?

Reply:
I'd say `is-active` but I suppose the service becomes inactive when it's done, right? So I don't believe we have good options here.

Reply:
Yeah, that's exactly the problem: it's a `Type=oneshot` with `RemainAfterExit=no` service, which means that after it runs it no longer is considered active by `systemd`.

Reply:
If you want to open an issue upstream in systemd, feel free :), otherwise we can just consider it acceptable here

Reply:
As tempting as that is I suspect this way of doing things will close the issue sooner than waiting for a new release of systemd.

Reply:
We can have both of them :)